lib/spack/spack/mirror.py

```diff
@@ -178,6 +180,7 @@ def to_json(self, stream=None):
     def to_yaml(self, stream=None):
         return syaml.dump(self.to_dict(True), stream)
 
+    # TODO: this isn't called anywhere
```

Comment:
@opadron this ended up being a red herring, I don't think this function is actually called anywhere, should it be removed?

Reply:
Sure.  For context, I had studied other Spack classes that encapsulated configuration and was just trying to add the same (de)serialization methods I found in those.  If they're not necessary I have no problem removing them.

Reply:
IMO it's best to leave out functions if they aren't called anywhere (for example in this case I got sidetracked thinking that's how `MirrorCollection` objects are normally constructed). It's not immediately relevant to the bug though so I'll leave it (and the TODO) for now - thanks!